pkg/console/subresource/route/route.go

```diff
@@ -162,31 +166,42 @@ func wildcard() routev1.WildcardPolicyType {
 	return routev1.WildcardPolicyNone
 }
 
+// The canonical host is the ingress on the route with a host that matches the
+// ingressConfig.Spec.Domain.  When we have an ingress that matches, and when it
+// is admitted, the console will be fully accessible.
+func GetCanonicalHost(route *routev1.Route, ingressConfig *configv1.Ingress) (string, error) {
+	// only checking canonical host against admitted ingress
+	for _, ingress := range route.Status.Ingress {
+		if !isIngressAdmitted(ingress) {
+			continue
+		}
+		if ingress.RouterCanonicalHostname == ingressConfig.Spec.Domain {
```

Comment:
Considering as well:

```golang
route.status.ingress[].routerName==default
```

Reply:
```golang
status:
  ingress:
  - conditions:
    - lastTransitionTime: 2019-04-01T15:07:00Z
      status: "True"
      type: Admitted
    host: console-openshift-console.apps.bpetersen-apr-1.devcluster.openshift.com
    routerCanonicalHostname: apps.bpetersen-apr-1.devcluster.openshift.com
    // this is probably the one we want, and means we may not need the ingressConfig at all.
    routerName: default
    wildcardPolicy: None```
```

Reply:
So here Im not sure:
1. If the `routerName` will be always `default`,  cause I haven't found any docs or code that would suggest that `default` is the value that we should be checking.

2. Not sure if we shouldn't check both, 
```go
if ingress.RouterName == default  && ingress.RouterCanonicalHostname == ingressConfig.Spec.Domain {
...
```

Reply:
Right, is the name `default` intended to be API.

Reply:
@smarterclayton is it okay for other components to rely on the `default` ingresscontroller name? I'm not sure what alternative we could suggest (a selector?).

Reply:
Decision is to trust that the instance named `default` is the OOTB ingresscontroller. So the name `default` is indeed implicit stable API.

Reply:
Great we will work with this then.